### ISSUE TITLE
Uninitialized constant Date::ZONES

### ISSUE THREAD

**Issue by Aupajo** [jeremyevans/home_run#3]
Using RVM Ruby 1.9.2p0 on OS X 10.6, I ran `gem install home_run` which reported being installed successfully, and then `home_run --bench`, which produced the following error: 

```
/Users/pete/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/rake.rb:2482:in `const_missing': uninitialized constant Date::ZONES (NameError)
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/gems/home_run-0.9.0/ext/date/format.rb:60:in `<module:Format>'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/gems/home_run-0.9.0/ext/date/format.rb:17:in `<class:Date>'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/gems/home_run-0.9.0/ext/date/format.rb:6:in `<top (required)>'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from /Users/pete/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/time.rb:39:in `<top (required)>'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/gems/rdoc-2.5.10/lib/rdoc/rdoc.rb:16:in `<top (required)>'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/gems/home_run-0.9.0/Rakefile:10:in `<top (required)>'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/gems/home_run-0.9.0/bin/home_run:39:in `load'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/gems/home_run-0.9.0/bin/home_run:39:in `<top (required)>'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/bin/home_run:19:in `load'
    from /Users/pete/.rvm/gems/ruby-1.9.2-p0/bin/home_run:19:in `<main>'
```

Also, through `irb` (RubyGems already loaded):

```
$ irb
ruby-1.9.2-p0 > require 'home_run'
LoadError: no such file to load -- home_run
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from (irb):1
    from /Users/pete/.rvm/rubies/ruby-1.9.2-p0/bin/irb:17:in `<main>'
ruby-1.9.2-p0 > 
```

**Comment by jeremyevans** [comment]
The "require 'home_run'" was added after the 0.9.0 gem release, so that's why that doesn't work.

The Rakefile was already changed to not require rdoc, so you shouldn't get this issue in the master branch.  As a workaround until the next gem, just delete the lines related to the rdoc task from the Rakefile.

**Comment by barttenbrinke** [comment]
This is still broken in the current master branch.

# home_run --install

cp date.rb /usr/local/lib/site_ruby/1.8
mkdir -p /usr/local/lib/site_ruby/1.8/date
cp date/format.rb /usr/local/lib/site_ruby/1.8/date
cp date_ext.so /usr/local/lib/site_ruby/1.8/x86_64-linux

# gem list

ERROR:  While executing gem ... (NameError)
    uninitialized constant Date::ZONES

# ruby -v

ruby 1.8.7 (2008-08-11 patchlevel 72) [x86_64-linux]

However, the --bench works fine

**Comment by jeremyevans** [comment]
I'm reopening this until we can get it sorted out.  Unfortunately, I can't recreate the issue:

$ sudo bin/home_run --install
Password:
cp date.rb /usr/local/lib/ruby/site_ruby/1.8
mkdir -p /usr/local/lib/ruby/site_ruby/1.8/date
cp date/format.rb /usr/local/lib/ruby/site_ruby/1.8/date
cp date_ext.so /usr/local/lib/ruby/site_ruby/1.8/i386-openbsd4.6
$ gem list

**\* LOCAL GEMS ***

...

Can you include the full backtrace?

**Comment by ghost** [comment]
Here's my console session from OSX 10.6.4, RVM/REE:

~: rvm info
ree-1.8.7-2010.02:

  system:
    uname:        "Darwin okokok.curbednetwork.com 10.4.0 Darwin Kernel Version 10.4.0: Fri Apr 23 18:28:53 PDT 2010; root:xnu-1504.7.4~1/RELEASE_I386 i386"
    shell:        "bash"
    version:      "3.2.48(1)-release"

  rvm:
    version:      "rvm 0.1.41 by Wayne E. Seguin (wayneeseguin@gmail.com) [http://rvm.beginrescueend.com/]"

  ruby:
    interpreter:  "ruby"
    version:      "1.8.7"
    date:         "2010-04-19"
    platform:     "i686-darwin10.3.0"
    patchlevel:   "2010-04-19 patchlevel 253"
    full_version: "ruby 1.8.7 (2010-04-19 patchlevel 253) [i686-darwin10.3.0], MBARI 0x6770, Ruby Enterprise Edition 2010.02"

  homes:
    gem:          "/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02"
    ruby:         "/Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02"

  binaries:
    ruby:         "/Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/bin/ruby"
    irb:          "/Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/bin/irb"
    gem:          "/Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/bin/gem"
    rake:         "/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/bin/rake"

  environment:
    PATH:         "/Users/eshepard/.rvm/bin:/Users/eshepard/.rvm/bin:/opt/local/bin:/opt/local/sbin:/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/bin:/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02@global/bin:/Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11/bin:/usr/local/sbin:/usr/local/mysql/bin"
    GEM_HOME:     "/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02"
    GEM_PATH:     "/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02:/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02@global"
    BUNDLE_PATH:  "/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02"
    MY_RUBY_HOME: "/Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02"
    IRBRC:        "/Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/.irbrc"
    RUBYOPT:      ""
    gemset:       ""

~: gem install home_run
Building native extensions.  This could take a while...
Successfully installed home_run-0.9.0
1 gem installed
~: home_run --install
cp date.rb /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8
mkdir -p /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/date
cp date/format.rb /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/date
cp date_ext.bundle /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/i686-darwin10.3.0
~: home_run --bench
/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/rake-0.8.7/lib/rake.rb:2503:in `const_missing': uninitialized constant Date::ZONES (NameError)
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/home_run-0.9.0/ext/date/format.rb:60
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/1.8/parsedate.rb:22
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/1.8/time.rb:43
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `require'
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/rdoc-2.5.9/lib/rdoc/rdoc.rb:16
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `require'
    from ./Rakefile:10
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/home_run-0.9.0/bin/home_run:39:in`load'
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/home_run-0.9.0/bin/home_run:39
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/bin/home_run:19:in `load'
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/bin/home_run:19
~: home_run --spec
/Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/rake-0.8.7/lib/rake.rb:2503:in`const_missing': uninitialized constant Date::ZONES (NameError)
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/home_run-0.9.0/ext/date/format.rb:60
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/1.8/parsedate.rb:22
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/1.8/time.rb:43
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`require'
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/rdoc-2.5.9/lib/rdoc/rdoc.rb:16
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
    from /Users/eshepard/.rvm/rubies/ree-1.8.7-2010.02/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`require'
    from ./Rakefile:10
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/home_run-0.9.0/bin/home_run:48:in `load'
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/gems/home_run-0.9.0/bin/home_run:48
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/bin/home_run:19:in`load'
    from /Users/eshepard/.rvm/gems/ree-1.8.7-2010.02/bin/home_run:19

**Comment by jeremyevans** [comment]
As I mentioned in the first comment, it's a known issue in the 0.9.0 gem.

As the backtrace shows, it tries to load rdoc, which loads time, which loads parsedate, which loads date/format.

The post 0.9.0 change was to not require rdoc, but that doesn't solve the root cause.  I think I've found the root cause, or at least part of it.  In date/format, I was only requiring 'date' if Date was not defined, so if any other code opened up the date class before requiring 'date/format', it would break.  I've now changed it to check for Date::ZONES instead of just Date.

Please try the master branch.  If you want to build a gem from the master branch, it's fairly easy:

```
rake gem
```

The only issue is that you need Ragel installed.

I'll leave this ticket open until I hear from you.

**Comment by ghost** [comment]
Sorry, missed the comment about building the gem from master. Thanks.

**Comment by barttenbrinke** [comment]
No problem :). I really would like to try this out! We are parsing 1000s of dates per request, so I'd like to see how this performs.

**Comment by jeremyevans** [comment]
I think this issue has been fixed by http://github.com/jeremyevans/home_run/commit/19ea2f83edabde0500d2c516c0077e6d6574df6b

If you still have a problem after building from the master branch, please reopen.

**Comment by barttenbrinke** [comment]
Will test tomorrow :)

**Comment by Aupajo** [comment]
Works great for me!

I uninstalled the gem, installed Ragel, built from master and installed the built gem.

Both `require 'home_run'` and `home_run --bench` now work fine.

**Comment by barttenbrinke** [comment]
Works under system ruby, doesn't work for me under rvm, eventhough I use rvm system.

**Comment by jeremyevans** [comment]
barttenbrinke: That certainly sounds like an rvm issue.  Can you paste the backtrace?  And are you using the master branch or the gem?

**Comment by Aupajo** [comment]
barttenbrinke: I'm using 1.9.2 with RVM and it worked for me.

**Comment by glaville** [comment]
That seems to cause problems with rubygem too :
(home_run 0.9.0/MRI 1.9.1p430/rubygem 1.3.7)

root@linux:~# gem update
ERROR:  Loading command: update (NameError)
    uninitialized constant Date::ZONES
ERROR:  While executing gem ... (NameError)
    uninitialized constant Gem::Commands::UpdateCommand
root@linux:~# gem list
ERROR:  Loading command: list (NameError)
    uninitialized constant Date::ZONES
ERROR:  While executing gem ... (NameError)
    uninitialized constant Gem::Commands::ListCommand

I'll test with master as soon as I can...

**Comment by jeremyevans** [comment]
Try the new 0.9.1 gem, that should fix it.

**Comment by brainopia** [comment]
I've got this problem using rails 3 and rvm with ruby 1.9.2.

$ rails c
/Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/home_run-0.9.2/lib/date/format.rb:60:in `module:Format': uninitialized constant Date::ZONES (NameError)

The problem appears only when using rails. If I install home_run into site_ruby then I've got another problem (again only with rails):

$ rails c
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant ITALY
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant ENGLAND
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant GREGORIAN
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant JULIAN
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant ZONES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant MONTHNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant ABBR_MONTHNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant DAYNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/x86_64-darwin10.4.0/date_ext.bundle: warning: already initialized constant ABBR_DAYNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/date.rb:2:in `require': undefined method`today' for `DateTime' (NameError)
    from /Users/brainopia/.rvm/rubies/ruby-1.9.2-rc2/lib/ruby/site_ruby/1.9.1/date.rb:2:in`<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/activesupport-3.0.0/lib/active_support/core_ext/date/calculations.rb:1:in `require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/activesupport-3.0.0/lib/active_support/core_ext/date/calculations.rb:1:in`<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/activesupport-3.0.0/lib/active_support/ruby/shim.rb:12:in `require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/activesupport-3.0.0/lib/active_support/ruby/shim.rb:12:in`<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/actionpack-3.0.0/lib/abstract_controller.rb:6:in `require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/actionpack-3.0.0/lib/abstract_controller.rb:6:in`<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/actionpack-3.0.0/lib/action_controller.rb:1:in `require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/actionpack-3.0.0/lib/action_controller.rb:1:in`<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/actionpack-3.0.0/lib/action_controller/railtie.rb:2:in `require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/actionpack-3.0.0/lib/action_controller/railtie.rb:2:in`<top (required)>'
    from /Users/brainopia/Code/rails/airlines/config/application.rb:5:in `require'
    from /Users/brainopia/Code/rails/airlines/config/application.rb:5:in`<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/railties-3.0.0/lib/rails/commands.rb:21:in `require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-rc2/gems/railties-3.0.0/lib/rails/commands.rb:21:in`<top (required)>'
    from script/rails:6:in `require'
    from script/rails:6:in`<main>'

**Comment by brainopia** [comment]
http://blade.nagaokaut.ac.jp/cgi-bin/scat.rb/ruby/ruby-core/4226 - my second error looks very similar to this

**Comment by jeremyevans** [comment]
It's possible to add the patch similar to that one, but it really masks the bigger issue, which is that date is being required twice when it shouldn't be.  Does it work correctly when you do `home_run rails c`?

**Comment by brainopia** [comment]
No, it doesn't. I've got the same error about undefined method "today".

**Comment by jeremyevans** [comment]
Try the patch at http://pastie.org/1168510.txt and let me know if it works.

**Comment by brainopia** [comment]
If I try to run as you advised then I have exception. But if I install home_run into site_ruby after patching it with your patch then everything is all right. 

$ home_run rails c
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date/format.rb:8: warning: already initialized constant MONTHS
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date/format.rb:14: warning: already initialized constant DAYS
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date/format.rb:19: warning: already initialized constant ABBR_MONTHS
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date/format.rb:25: warning: already initialized constant ABBR_DAYS
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date/format.rb:30: warning: already initialized constant ZONES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:236: warning: already initialized constant MONTHNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:242: warning: already initialized constant DAYNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:245: warning: already initialized constant ABBR_MONTHNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:249: warning: already initialized constant ABBR_DAYNAMES
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:301: warning: already initialized constant ITALY
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:305: warning: already initialized constant ENGLAND
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:309: warning: already initialized constant JULIAN
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:313: warning: already initialized constant GREGORIAN
/Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:1843:in `private_class_method': undefined method`today' for class `Class' (NameError)
    from /Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:1843:in`class:DateTime'
    from /Users/brainopia/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/date.rb:1829:in `<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/activesupport-3.0.0/lib/active_support/core_ext/date/calculations.rb:1:in`require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/activesupport-3.0.0/lib/active_support/core_ext/date/calculations.rb:1:in `<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/activesupport-3.0.0/lib/active_support/ruby/shim.rb:12:in`require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/activesupport-3.0.0/lib/active_support/ruby/shim.rb:12:in `<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/actionpack-3.0.0/lib/abstract_controller.rb:6:in`require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/actionpack-3.0.0/lib/abstract_controller.rb:6:in `<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/actionpack-3.0.0/lib/action_controller.rb:1:in`require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/actionpack-3.0.0/lib/action_controller.rb:1:in `<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/actionpack-3.0.0/lib/action_controller/railtie.rb:2:in`require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/actionpack-3.0.0/lib/action_controller/railtie.rb:2:in `<top (required)>'
    from /Users/brainopia/Code/rails/airlines/config/application.rb:5:in`require'
    from /Users/brainopia/Code/rails/airlines/config/application.rb:5:in `<top (required)>'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/railties-3.0.0/lib/rails/commands.rb:21:in`require'
    from /Users/brainopia/.rvm/gems/ruby-1.9.2-p0/gems/railties-3.0.0/lib/rails/commands.rb:21:in `<top (required)>'
    from script/rails:6:in`require'
    from script/rails:6:in `<main>'

**Comment by brainopia** [comment]
Hm, does home_run implement day_fraction_to_time, cause I've got exception undefined_method `day_fraction_to_time', although there is one in documentation http://rdoc.info/github/jeremyevans/home_run/master/toplevel#day_fraction_to_time-instance_method.

**Comment by jeremyevans** [comment]
day_fraction_to_time is a private method on 1.9, and home_run doesn't implement the private methods.  The reason it shows up on the rdoc is that it's a public method on 1.8, and home_run implements it on 1.8.

The supported ways of using home_run are either via `home_run --install` or `home_run program args`.  You get the error when you don't install it because it is using the stdlib date class instead of the home_run date class.

**Comment by jeremyevans** [comment]
I've pushed the patch at http://pastie.org/1168510.txt and released a new gem (version 0.9.3).
